src/Peachpie.CodeAnalysis/Utilities/DistinctQueue.cs

```diff
@@ -5,18 +5,27 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using Pchp.CodeAnalysis.Semantics.Graph;
+using Peachpie.CodeAnalysis.Utilities;
 
 namespace Pchp.CodeAnalysis.Utilities
 {
     /// <summary>
-    /// Represents queue where items are enqueued just once and queue can be accessed in parralel.
+    /// Represents priority queue where items are enqueued just once and queue can be accessed in parallel.
     /// </summary>
-    internal sealed class DistinctQueue<T>
+    internal sealed class DistinctQueue<T> where T : BoundBlock
```

Comment:
I would remove the condition (it makes `DistinctQueue` unusable for anything else).